contracts/truefi3/interface/IDebtInstrument.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+/**
+ * @title IDebtInstrument
+ * @notice Represents debt in the TrueFi protocol
+ * @dev This is the basic interface for representing Debt in TrueFi
+ * This interface is meant to be extended to support different debt types
+ *
+ * An NFT represents some debt where:
+ *  - Each debt has an owner and a borrower
+ *  - The owner mints and funds a loan
+ *  - The borrower can withdraw funds from the loan
+ *  - The borrower calls repay() to pay back the loan
+ *  - Repayments are withdrawable by the owner
+ *  - Every DebtInstrument has an underlying token
+ *  - Only principal debt is tracked
+ *
+ **/
+interface IDebtInstrument is IERC721 {
+    /// @dev The owner address lends tokens and withdraws repayments
+    function owner() external view returns (address);
+
+    /// @dev The borrower address borrows and repays the loan
+    function borrower() external view returns (address);
+
+    /// @dev The underlying token denominating the debt
+    function token() external view returns (address);
+
+    /// @dev Principal debt amount
+    function principal() external view returns (uint256);
+
+    /// @dev Each debt has an expiration timestamp
+    function expiry() external view returns (uint256);
+
+    /// @dev Borrower calls this function to borrow `amount` of tokens
+    function borrow(uint256 amount) external;
+
+    /// @dev Borrower calls this function to repay `amount` of tokens
+    function repay(uint256 amount) external;
+
+    /// @dev Owner funds a loan for `amount` and increases principal
+    function fund(uint256 amount) external;
+
+    /// @dev Owner calls this function to withdraw `amount` of tokens
+    function withdraw(uint256 amount) external;
```

Comment:
Does `withdraw()` reduce `principal` by amount? Otherwise we'll also want to track and publish `withdrawn()` or `repaid()` views so that repayment state doesn't get lost.

This is similar to `tokenRedeemed` in our current FTLs.

Reply:
Good question. I think we have to track the amount repaid. I'm trying *not* to have any functions regarding interest involved here.